mbed-hal/spi_api.h

```diff
@@ -64,43 +97,64 @@ extern "C" {
 /** Initialize the SPI peripheral
  *
  * Configures the pins used by SPI, sets a default format and frequency, and enables the peripheral
+ * @note The function asserts that the selected pins map onto one SPI peripheral!
+ *
  * @param[out] obj  The SPI object to initialize
  * @param[in]  mosi The pin to use for MOSI
  * @param[in]  miso The pin to use for MISO
  * @param[in]  sclk The pin to use for SCLK
+ *
+ * @retval  `SPI_RC_OK`
  */
-void spi_init(spi_t *obj, PinName mosi, PinName miso, PinName sclk);
+spi_return_code_t spi_init(spi_t *obj, PinName mosi, PinName miso, PinName sclk);
```

Comment:
There is another idea floating around with a complete change to this API, including proper CS support.

The current proposal is simple to implement and makes it easier to unit test the current implementations.
(instead of just silently failing, [like here](https://github.com/ARMmbed/mbed-hal-ksdk-mcu/blob/master/source/spi_api.c#L165-L185)).